lib/IRGen/GenCast.cpp

```diff
@@ -1089,6 +1090,17 @@ llvm::Value *irgen::emitFastClassCastIfPossible(IRGenFunction &IGF,
 
   // Get the metadata pointer of the destination class type.
   llvm::Value *destMetadata = IGF.IGM.getAddrOfTypeMetadata(targetFormalType);
```

Comment:
It seems that we should no longer call `getAddrOfTypeMetadata` for lazy initialized metadata. It looks like dead code to me.

Reply:
No, it is not dead code - it is simply unnecessary work.  It is the "else" case that is being hoisted and done unconditionally.  I can sink it into an `else` clause.